packages/datagateway-common/src/api/retryICATErrors.ts

```diff
@@ -1,17 +1,45 @@
 import { AxiosError } from 'axios';
+import { useQueryClient } from 'react-query';
 
-const retryICATErrors = (failureCount: number, error: AxiosError): boolean => {
+const createRetryICATErrors = (
+  retries: number
+): ((failureCount: number, error: AxiosError) => boolean) => {
+  return (failureCount: number, error: AxiosError) =>
+    baseRetryICATErrors(failureCount, error, retries);
+};
+
+const baseRetryICATErrors = (
+  failureCount: number,
+  error: AxiosError,
+  retries: number
+): boolean => {
   const message =
     (error as AxiosError<{ message?: string }>).response?.data?.message ??
     error.message;
   if (
     error.response?.status === 403 ||
     // TopCAT doesn't set 403 for session ID failure, so detect by looking at the message
     message.toUpperCase().includes('SESSION') ||
-    failureCount >= 3
+    failureCount >= retries
   )
     return false;
   return true;
 };
 
-export default retryICATErrors;
+export const useRetryICATErrors = (): ((
+  failureCount: number,
+  error: AxiosError
+) => boolean) => {
+  const queryClient = useQueryClient();
+  const opts = queryClient.getDefaultOptions();
+  // TODO: do we want to be more elegant in handling other types of retry...
```

Comment:
Is this todo still needed?  I think this is fine, since as you say 0 is the same as false here, and it looks like the only other possible type is a function which we can't set in the config, unless I am missing something?

Reply:
Yeah I wrote this TODO before I handled false explicitly, and handling true makes little sense (unless I set it to infinity) and same with functions. Plus assuming retry of 3 is what the old function did. I'll remove the TODO